Source/ParleyTests/MessagesManagerTests.swift

```diff
@@ -0,0 +1,274 @@
+        
+        let todayMessageText = "Good morning!"
+        let todayMessageDate = Date()
+        let todayMessage = createUserMessage(todayMessageText, date: todayMessageDate)
+        _ = messagesManager.add(todayMessage)
+        
+        XCTAssertEqual(messagesManager.messages[0].message, MESSAGE_WELCOME_TEXT, "First message should be welcome message.")
+        
+        XCTAssertEqual(messagesManager.messages[1].type, .date, "Second message should be date header.")
+        XCTAssertEqual(messagesManager.messages[1].message, yesterdayDate.asDate(), "Date indicator text should be the same as the formatted date.")
+        
+        XCTAssertEqual(messagesManager.messages[2].message, yesterdayMessageText, "Yesterday message should have the correct message text.")
+        XCTAssertEqual(messagesManager.messages[2].type, .user, "Yesterday message should be from the user.")
+        
+        XCTAssertEqual(messagesManager.messages[3].type, .date, "Fourth message should be date header.")
+        XCTAssertEqual(messagesManager.messages[3].message, todayMessageDate.asDate(), "Date indicator text should be the same as the formatted date.")
+
+        XCTAssertEqual(messagesManager.messages[4].message, todayMessageText, "Today message should have the correct message text.")
+        XCTAssertEqual(messagesManager.messages[4].type, .user, "Today message should be from the user.")
+    }
+    
+    func testMessage_ShouldBeIgnored_WhenTryingToAddTheSameMessageTwice() {
+        let message = createUserMessage("Good morning")
+        let firstInsertionIndexPaths = messagesManager.add(message)
+        
+        XCTAssertEqual(firstInsertionIndexPaths.count, 2, "There should be two IndexPath after inserting a new message. One for the date indicator and the other is for the new message.")
+        XCTAssertEqual(messagesManager.messages.count, 2, "There should now be two one message in the messages array.")
+        
+        let secondInsertionIndexPaths = messagesManager.add(message)
+        XCTAssertTrue(secondInsertionIndexPaths.isEmpty, "Array should be empty because it should not insert the same message twice.")
+        XCTAssertEqual(messagesManager.messages.count, 2, "Array of messages should remain the same.")
+    }
+    
+    func testTypingMessage_ShouldBeLastMessage_WhenAddingTheTypingMessage() {
+        addInfoMessageAndFirstUserMessage()
+        
+        let indexPaths = messagesManager.addTypingMessage()
+        XCTAssertEqual(indexPaths.count, 1, "Adding Typing indicator should only add a single IndexPath.")
+        XCTAssertEqual(indexPaths[0].row, 3, "Typing message should be the fourth index.")
+        XCTAssertEqual(indexPaths[0].section, 0, "Typing message should be the fourth index.")
+        
+        XCTAssertEqual(messagesManager.messages.last!.type, .agentTyping, "Last message should be of type agent typing.")
+    }
+    
+    func testAddTypingIndicator_ShouldBeIgnored_WhenTryingToAddItTwice() {
```

Comment:
component shouldn't have `Add` (can be omitted here since its part of the "when" part)